### ISSUE TITLE
[4.1] Relations with same name as field in database. 

### ISSUE THREAD

**Issue by Athuli7** [Laravel-Backpack/CRUD#2911]
# Bug report

### What I did

Upgraded to 4.1

### What I expected to happen

Things to work after the changes were made.

### What happened

Relations are messed up. Everything triggers a ```Call to undefined method Illuminate\Database\Eloquent\Builder::getRelated()``` error.

### What I've already tried to fix it

Everything I know. But there seems to no logic to the madness.

Initially, this error was seen when 'relationship' or 'select2' field used a relation that overrides the foreign key. It was fixed by running migrations to change to the default value.

But some other model didn't have the issue. And I haven't been able to root out the problem.

**Not working :-**
```
    public function ICOStatus() {
        return $this->belongsTo('App\Models\Generic', 'icostatus');
    }
    public function JudgmentStatus() {
        return $this->belongsTo('App\Models\Generic', 'jugdmentstatus');
    }
```

**Working :-**
```
    public function ActType()
    {
        return $this->belongsTo('App\Models\Generic', 'type');
    }
```

Now I have non-relationship fields like this throwing that error. But only if the field name is ```on```.

```
            [
                'name' => 'on',
                'label' => 'Date &amp; Time',
                'type' => 'datetime_picker',
                'datetime_picker_options' => [
                    'format' => 'DD/MM/YYYY HH:mm',
                    'language' => 'en'
                ]
            ],
```

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

**Comment by pxpm** [comment]
Hello @Athuli7 

Thanks for bring this up. Well, yesterday I said it would not be very common for people to use the same name of the relation as a field in database. 

Usually and per convention, you would have a `category_id` field in the database for a `category` relation, and not a `category` field in database. 

I wrote a more descriptive answer in here: 
https://github.com/Laravel-Backpack/CRUD/issues/2885#issuecomment-637393362

One solution would be to change the names of your relations in case changing the database field name is not available. 

Eg: 
```php
public function ICOStatusRelation() {
        return $this->belongsTo('App\Models\Generic', 'icostatus');
    }
```
and change the entity from `ICOStatus` to `ICOStatusRelation` in your field definition. 


The second error with using `on` as a name has been fixed #2862 

Let me know.

Best,
Pedro

**Comment by pxpm** [comment]
Closing this, some time as passed since last iteration and we have another open issue with more info on this subject.

If this stills an issue I suggest you comment out all your fields and try enabling one by one and see which one is the culprit, it might make the debug of the issue easier. Feel free to add comments to this thread if you have more findings and still think this is a bug.

Best,
Pedro
